src/index/ExternalSortFunctors.h

```diff
@@ -47,41 +47,22 @@ struct SortTriple {
       return cGraph < 0;
     }
   }
-
-  // Value that is strictly smaller than any input element.
-  static T min_value() { return {Id::min(), Id::min(), Id::min()}; }
-
-  // Value that is strictly larger than any input element.
-  static T max_value() { return {Id::max(), Id::max(), Id::max()}; }
 };
 
 using SortByPSO = SortTriple<1, 0, 2>;
 using SortByPSONoGraphColumn = SortTriple<1, 0, 2, false>;
 using SortBySPO = SortTriple<0, 1, 2>;
 using SortByOSP = SortTriple<2, 0, 1>;
 
-// TODO<joka921> Which of those are actually "IDs" and which are something else?
 struct SortText {
-  using T = std::tuple<TextBlockIndex, TextRecordIndex, WordOrEntityIndex,
-                       Score, bool>;
+  using Row = IdTableStatic<5>::row_type;
   // comparison function
-  bool operator()(const T& a, const T& b) const {
-    auto permute = [](const T& x) {
-      using namespace std;
-      return tie(get<0>(x), get<4>(x), get<1>(x), get<2>(x), get<3>(x));
+  bool operator()(const Row& a, const Row& b) const {
+    auto permute = [](const Row& x) {
+      return std::tie(x[0], x[4], x[1], x[2], x[3]);
```

Comment:
You Unfortunately shouldn't use the simple `tie`trick here, but something similar to the sorters above, where don't use `operator<` but `compareWithoutLocalVocab`, as this has a serious impact (20 percent or so) on the sorting performance.